### PR TITLE
feat: improved watcher

### PR DESCRIPTION
**Pull request by b0dhidharma**

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -26,20 +27,18 @@ module.exports = {
...
-          accounts: [process.env.KOVAN_PRIVATE_KEY, '0x8901af9255b653e9a8f654d84b53d37b9134eb5e949f394c8ddb0c2ef4481287'],
+        rinkeby: {
+          url: process.env.RINKEBY_HTTPS_URL,
+          accounts: [process.env.RINKEBY_PRIVATE_KEY, process.env.RINKEBY_2_PRIVATE_KEY],
           gasPrice: 'auto',
         },
-        goerli: {
-          url: process.env.GOERLI_HTTPS_URL,
-          accounts: [process.env.GOERLI_PRIVATE_KEY],
-          gasPrice: 'auto',
```
Comment (lbertenasco) [inline hardhat.config.ts]: :'(

File: scripts/sample-tx.ts [hunk 2]
```diff
@@ -1,33 +1,21 @@
...
-
 async function execute() {
   const [deployer] = await ethers.getSigners();
-  // const stealthVault = await ethers.getContractAt('contracts/StealthVault.sol:StealthVault', '0x12F86457C6aa1d0e63aef72b4E2ae391A7EeB14D');
-  const stealthRelayer = await ethers.getContractAt('contracts/StealthRelayer.sol:StealthRelayer', '0x4A7a3b790D0aD2b9e1e65f9a3cf31e99455D4E1c');
-  const stealthERC20 = await ethers.getContractAt('contracts/mock/StealthERC20.sol:StealthERC20', '0xf244E372A492e415599452b4eA139338f3f24a0b');
-  // await stealthRelayer.setPenalty(utils.parseEther('0.001'));
-  // await stealthVault.addStealthContract(stealthRelayer.address);
+  const stealthRelayer = await ethers.getContractAt('contracts/StealthRelayer.sol:StealthRelayer', '0x03900f1cdEf9355a121D84DeaC414799dB51Dc05');
+  const stealthERC20 = await ethers.getContractAt('contracts/mock/StealthERC20.sol:StealthERC20', '0x01489f5881A4C436793cb48434eAA2D488D83C07');
```
Comment (lbertenasco) [inline scripts/sample-tx.ts:12]: Addresses can be found in `utils/contracts.ts`

File: scripts/watcher/yearn.ts [hunk 3]
```diff
@@ -1,17 +1,83 @@
...
 import { BigNumber, Contract, utils, Transaction as EthersTransaction } from 'ethers';
-import { createAlchemyWeb3 } from '@alch/alchemy-web3';
 import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import Web3 from 'web3';
+import WebSocket from 'ws';
 
-// Using WebSockets
-const web3 = createAlchemyWeb3('wss://eth-kovan.ws.alchemyapi.io/v2/OAh_8Jbu8aMsuFAj1n8gRzo8PPRfK7VP');
+const web3 = new Web3('ws://127.0.0.1:8546');
+// const web3 = new Web3('wss://eth-rinkeby.ws.alchemyapi.io/v2/nS8CFVW0Srf7kOWvYG_fcPm18CXh66Kz');
```
Comment (lbertenasco) [inline scripts/watcher/yearn.ts]: remove key, and possibly deprecate from alchemy too

File: hardhat.config.ts (hunk 1 above)
Comment (b0dhidharma) [reply hardhat.config.ts]: Oops ! I can add it back again !

**Pull request merged**